packages/rocketchat-message-pin/server/pinMessage.js

```diff
@@ -42,16 +42,20 @@ Meteor.methods({
 		originalMessage = RocketChat.callbacks.run('beforeSaveMessage', originalMessage);
 		RocketChat.models.Messages.setPinnedByIdAndUserId(originalMessage._id, originalMessage.pinnedBy, originalMessage.pinned);
 
-		return RocketChat.models.Messages.createWithTypeRoomIdMessageAndUser('message_pinned', originalMessage.rid, '', me, {
-			attachments: [
-				{
-					'text': originalMessage.msg,
-					'author_name': originalMessage.u.username,
-					'author_icon': getAvatarUrlFromUsername(originalMessage.u.username),
-					'ts': originalMessage.ts
-				}
-			]
-		});
+		const attachments = [{
+			'text': originalMessage.msg,
+			'author_name': originalMessage.u.username,
+			'author_icon': getAvatarUrlFromUsername(originalMessage.u.username),
+			'ts': originalMessage.ts
+		}];
+
+		if (originalMessage.attachments !== undefined) {
+			originalMessage.attachments.forEach((attachment) => {
```

Comment:
what about changing this from merging all attachments into a single one to appending all attachments to the message?

Reply:
@sampaiodiego I've made the change.

Reply:
@c0dzilla thx, but I don't think it works..

I'd rather do it this way:

```
if (Array.isArray(originalMessage.attachments)) {
	attachments.push(...originalMessage.attachments);
}
```

this way `attachments` will receive all attachments from `originalMessage`.

this might not respect `Message_QuoteChainLimit` though.. so additional code is required.